Source/UI/Coordinators/PasswordCoordinator.swift

```diff
@@ -139,12 +139,24 @@ extension PasswordCoordinator {
             switch result {
             case let .success(currentUser):
                 self?.configuration.tracker?.loginID = currentUser.legacyID
-                self?.updatekeyChain(
-                    for: identifier,
-                    loginFlowVariant: loginFlowVariant,
-                    password: password
-                ) {
-                    self?.spawnCompleteProfileCoordinator(for: .signin(user: currentUser), persistUser: persistUser, completion: completion)
+                let device = UserDevice(
+                    hash: nil,
+                    applicationName: self?.configuration.appName,
+                    applicationVersion: self?.configuration.appVersion
+                )
+                currentUser.device.update(device) { result in
+                    switch result {
+                    case .success:
+                        self?.updatekeyChain(
+                            for: identifier,
+                            loginFlowVariant: loginFlowVariant,
+                            password: password
+                        ) {
+                            self?.spawnCompleteProfileCoordinator(for: .signin(user: currentUser), persistUser: persistUser, completion: completion)
+                        }
+                    case let .failure(error):
+                       self?.present(error: error)
```

Comment:
Does the login process have to stop if the device can't be updated? Should the update device call be fire and forget?

Reply:
No this is not supposed to fail.